File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-8.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        if nums[0] < nums[-1]:
            return nums[0]
        i = 1
        j = len(nums) - 1
        while i <= j:
            k = (i + j) // 2
            if nums[k - 1] > nums[k]:
                return nums[k]

            if nums[k] < nums[j]:
                j = k
            elif nums[i] < nums[k - 1]:
                i = k
            else:
                i += 1

        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-8.cpp
#include <algorithm> // std::sort
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::sort(nums.begin(), nums.end());
        int i = 0;
        int j = nums.size() - 1;
        while (i < j) {
            const int s = nums[i] + nums[j];
            if (s > target) {
                j -= 1;
            } else if (s < target) {
                i += 1;
            } else {
                break;
            }
        }
        vector<int> m = {i, j};
        return m;
    }
};
